src/io/obf.rs

```diff
@@ -220,17 +221,39 @@ pub async fn obfuscate<M, SU, SH, ST, R, P>(
             {
                 player.play_music(format!("bgm/obf_bgm{}.mp3", (2 * level + num) % 3 + 2));
             }
-            let r = &public_data.r[num];
-            log_mem(format!("Computed R ({},{}) (n+1)x(n+1)", r.row_size(), r.col_size()));
+
+            let tag = format!("S_{}_{}", level, num).into_bytes();
+            let s_i_bar =
```

Comment:
The reason of using the hash sampler is that just starting from the `key` and the `tag` someone could sample the same matrix. Indeed in the previous implementation the obfuscator would use the hash sampler for sampling the public keys and this would be reused by the evaluator. 

For the secret key, there's no specific reason to use the hash sampler as there's no "reusage" logic. So I suggest using the standard sampler, if faster.

Reply:
standard sampler means, uniform sampler with finite ring dist?

Reply:
yes

Reply:
The secret matrices must be generated with the unifrom sampler; otherwise, the evaluator can recompute them using the given hash key.

Reply:
Why does evaluator have ability to recompute if its secret?